hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/auth/AuthenticationService.java

```diff
@@ -0,0 +1,247 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.auth;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.websocket.CloseReason;
+import javax.websocket.CloseReason.CloseCodes;
+import javax.websocket.Session;
+
+import org.hawkular.cmdgw.api.AuthMessage;
+import org.hawkular.cmdgw.command.ws.WsEndpoints;
+import org.hawkular.cmdgw.command.ws.server.HttpRequestData;
+import org.hawkular.cmdgw.log.GatewayLoggers;
+import org.hawkular.cmdgw.log.MsgLogger;
+
+import com.squareup.okhttp.ConnectionPool;
+import com.squareup.okhttp.Credentials;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+/**
+ * A service that periodically re-authenticates all open {@link Session}s by forwarding their latest known credentials
+ * to {@link AuthenticationServlet}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@ApplicationScoped
+public class AuthenticationService {
+
+    private static final MsgLogger log = GatewayLoggers.getLogger(AuthenticationService.class);
+
+    /**
+     * See https://tools.ietf.org/html/rfc6455#section-5.5
+     */
+    public static final int MAX_REASON_LENGTH_UTF8_BYTES = 123;
+
+    public static final CloseReason MISSING_CREDENTIALS = new CloseReason(CloseCodes.UNEXPECTED_CONDITION,
+            "Missing " + org.hawkular.cmdgw.auth.Credentials.class.getSimpleName());
+    public static final CloseReason MISSING_HTTP_REQUESTDATA =
+            new CloseReason(CloseCodes.UNEXPECTED_CONDITION, "Missing " + HttpRequestData.class.getSimpleName());
+    public static final CloseReason NOT_AUTH_MESSAGE =
+            new CloseReason(CloseCodes.UNEXPECTED_CONDITION, "Expected " + AuthMessage.class.getSimpleName());
+
+    /** 5 minutes */
+    private static final String REAUTH_INTERVAL_DEFAULT = "300";
+
+    private static final String REAUTH_INTERVAL_PROP = "org.hawkular.cmdgw.reauthIntervalSeconds";
+
+    /** A custom {@link CloseReason} that we use to signal success of the re-authentication */
```

Comment:
Not sure why you need this. This way, you are assuming that the client and server know about this special `CloseReason`, which is not always the case (should not, at least). Why don't you just close with code `1000`? As per the RFC:

      1000 indicates a normal closure, meaning that the purpose for
      which the connection was established has been fulfilled.